service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -285,4 +296,52 @@ public String getRequiredGcpProject(UUID workspaceId) {
         .getGcpCloudContext()
         .map(GcpCloudContext::getGcpProjectId);
   }
+
+  /**
+   * Grant a user permission to impersonate their pet service account in a given workspace. Unlike
+   * other operations, this does not run in a flight because it only requires one write operation.
+   * This operation is idempotent.
+   *
+   * @return The email identifier of the user's pet SA in the given workspace.
+   */
+  public String enablePet(UUID workspaceId, AuthenticatedUserRequest userRequest) {
+    final String serviceAccountUserRole = "roles/iam.serviceAccountUser";
+    Workspace workspace =
+        validateWorkspaceAndAction(
+            userRequest, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
+    stageService.assertMcWorkspace(workspace, "enablePet");
+
+    String userEmail =
+        SamService.rethrowIfSamInterrupted(
+            () -> samService.getRequestUserEmail(userRequest), "getUserEmail");
+    String projectId = getRequiredGcpProject(workspaceId);
+    String petSaEmail = samService.getPetSaEmail(projectId, userRequest);
+    ServiceAccountName petSaName =
+        ServiceAccountName.builder().email(petSaEmail).projectId(projectId).build();
+    try {
+      Policy saPolicy =
+          crlService.getIamCow().projects().serviceAccounts().getIamPolicy(petSaName).execute();
+      Binding saUserBinding =
+          new Binding()
+              .setRole(serviceAccountUserRole)
+              .setMembers(Collections.singletonList("user:" + userEmail));
+      // If no bindings exist, getBindings() returns null instead of an empty list.
```

Comment:
I hit another API that does the same thing. So annoying.

Reply:
Yeah, kind of a weird quirk from GCP